### DISCUSSION TITLE
Exitcode=1 for BMP files which length and filesize do not match

### DISCUSSION THREAD

**Discussion by qliu67** [ImageMagick/ImageMagick#3634] [Development]
I have some BMP files which failed ImageMagick inspection due to length and filesize do not match.
The convert command will generate the correct output jpg, but still shows exitcode=1.
Should we treat it as a warning and make exitcode=0? I don't want to worry too much about this warning, but I do care about the exitcode. 

This happens on ImageMagick 7.0.11-2, running on a Linux container:
```
UNKNOWN_NODE_NAME $ convert input.BMP output.jpg
convert: length and filesize do not match `input.BMP' @ error/bmp.c/ReadBMPImage/845.
UNKNOWN_NODE_NAME $ echo $?
1
```
However the exitcode is 0 on my mac which runs an older version: ImageMagick 7.0.9-5
```
Downloads $ convert input.BMP output.jpg
convert: length and filesize do not match `input.BMP' @ warning/bmp.c/ReadBMPImage/846.
Downloads $ echo $?
0
```

https://www.dropbox.com/s/d62vb6fidyw9iqx/LAND2.BMP?dl=0

Here is a sample file I find from Internet, which shows the same issue when I run those commands. I think it's affecting a variety of BMP files. I hope these's an option to allow it to be exitcode=0.

FYI this might relate to https://github.com/dlemstra/Magick.NET/issues/528

**Comment by snibgo** [comment]
I suggest you provide a link to your BMP file. Then a developer can take a look.

**Comment by qliu67** [comment]
https://www.dropbox.com/s/d62vb6fidyw9iqx/LAND2.BMP?dl=0

Here is a sample file I find from Internet, which shows the same issue when I run those commands. I think it's affecting a variety of BMP files. I hope these's an option to allow it to be exitcode=0.

FYI this might relate to https://github.com/dlemstra/Magick.NET/issues/528

**Comment by snibgo** [comment]
You say:
> The convert command will generate the correct output jpg, but still shows exitcode=1

... but are you sure? Perhaps the output file was already there, from a successful command.

With IM v7.0.10-45 on Windows 8.1, the command:
```
magick land2.bmp x.png
```
... gives the error ...
```
magick: length and filesize do not match `land2.bmp' @ error/bmp.c/ReadBMPImage/845.
```
... and does _not_ create an output file.

In general, IM has become more secure over time, more defensive against files that do not conform to standards. IM raises an error, then exits with no further processing.

**Reply by qliu67** [reply]
The magick command won't generate an output file. But convert command will generate a correct output.
```
$ convert LAND2.BMP x.jpg
convert: length and filesize do not match `LAND2.BMP' @ error/bmp.c/ReadBMPImage/845.
```
I'm running the latest version on mac:
```
$ convert --version
Version: ImageMagick 7.0.11-10 Q16 x86_64 2021-04-29 https://imagemagick.org
```
For both cases, the exitcode is 1.
```
$ echo $?
1
```

**Comment by dlemstra** [comment]
Snibgo is correct. Earlier version would raise a warning under certain conditions but we changed this to always raise an error. And later we introduced an option to ignore this check: `magick -define bmp:ignore-filesize=true input.BMP output.jpg`

**Reply by qliu67** [reply]
Thanks! Glad we have this option! I have verified it worked just fine for our test files, successfully converted them with exitcode=0.
We have many BMP images produced from some inspection machines, which don't meet the standard for other normal images. This option helps a lot.

**Comment by qliu67** [comment]
How does ImageMagick check the length and file size? I want to make sure that the example file does have different file size and length. Or is it a bug how ImageMagick do the checks? Could you help to tell me how to verify that the example file I shared has different file sizes? @dlemstra @snibgo

**Reply by qliu67** [reply]
When I run magick identify command or exiftool command, it shows the size is 787510.
